common/src/main/java/com/wynntils/features/user/WorldWaypointDistanceFeature.java

```diff
@@ -78,10 +79,19 @@ public void onRenderLevelPost(RenderLevelEvent.Post event) {
         projection.multiply(Vector3f.XP.rotationDegrees(camera.getXRot()));
         projection.multiply(Vector3f.YP.rotationDegrees(camera.getYRot() + 180.0F));
 
+        float dx;
+        float dy;
+        float dz;
         // offset to put text to the center of the block
-        float dx = (float) (location.x + 0.5 - cameraPos.x);
-        float dy = (float) (location.y + 0.5 - cameraPos.y);
-        float dz = (float) (location.z + 0.5 - cameraPos.z);
+        if (location.y <= 0 || location.y > 255) {
+            dx = (float) (location.x + 0.5 - cameraPos.x);
+            dy = (float) (cameraPos.y + 0.5 - cameraPos.y);
+            dz = (float) (location.z + 0.5 - cameraPos.z);
+        } else {
+            dx = (float) (location.x + 0.5 - cameraPos.x);
+            dy = (float) (location.y + 0.5 - cameraPos.y);
+            dz = (float) (location.z + 0.5 - cameraPos.z);
+        }
```

Comment:
You can change this code so you only calculate dy in the if statements, and the other at declaration.